engine/class_modules/sc_death_knight.cpp

```diff
@@ -8907,11 +8908,23 @@ void death_knight_t::default_apl_blood()
   // Cooldowns
   def -> add_action( this, "Vampiric Blood" );
   def -> add_action( "potion,if=buff.dancing_rune_weapon.up" );
-  def -> add_action( this, "Dancing Rune Weapon", "if=!talent.blooddrinker.enabled|!cooldown.blooddrinker.ready" );
-  def -> add_talent( this, "Tombstone", "if=buff.bone_shield.stack>=7" );
+  //def -> add_action( "call_action_list,name=covenants" );
   def -> add_action( "call_action_list,name=essences" );
   def -> add_action( "call_action_list,name=standard" );
-
+  /*
+  // Hardcast Death's Due with no CS proc if there is < 3s on the buff.
+  covenants -> add_action( "deaths_due,if=covenant.night_fae&(!buff.deaths_due.up|buff.deaths_due.remains<5|buff.crimson_scourge.up)" );
+  // Cast HS on the last tick of Death's Due to refresh the buff as late as possible
+  covenants -> add_action( "heart_strike,if=covenant.night_fae&death_and_decay.remains<3" );
```

Comment:
```suggestion
  covenants -> add_action( "death_strike,if=covenant.night_fae&death_and_decay.remains<5&runic_power.deficit<=(15+buff.dancing_rune_weapon.up*5+spell_targets.heart_strike*talent.heartbreaker.enabled*2)" );
  covenants -> add_action( "heart_strike,if=covenant.night_fae&death_and_decay.remains<3" );
  covenants -> add_action( "heart_strike,if=covenant.night_fae&death_and_decay.ticking&buff.deaths_due.remains<3");
```

Original implementation was an actual dps loss.  The buff was dropping as it wasn't heart striking after dropping a new DD.  After adding that it it lead to resource flooding issues, so runic needs to be burned off before a HS near the end of DD to keep the buff rolling.